asdf_astropy/converters/coordinates/frame.py

```diff
@@ -0,0 +1,86 @@
+from asdf.extension import Converter
+
+from ..utils import import_type
+
+
+class FrameConverter(Converter):
+    def __init__(self, tag, frame_type_name):
+        self._tag = tag
+        self._frame_type_name = frame_type_name
+        self._frame_type = None
+
+    @property
+    def tags(self):
+        return [self._tag]
+
+    @property
+    def types(self):
+        return [self._frame_type_name]
+
+    @property
+    def frame_type(self):
+        # Delay import until the frame class is needed to improve speed
+        # of loading the extension.
+        if self._frame_type is None:
+            self._frame_type = import_type(self._frame_type_name)
+        return self._frame_type
+
+    def to_yaml_tree(self, obj, tag, ctx):
+        node = {}
+
+        if obj.has_data:
+            node["data"] = obj.data
+
+        # TODO: Figure out why we can't use the frame_attributes
```

Comment:
It looks to me like frame_attributes is a dict of attribute name to attribute value, yet for some reason the original tag class calls getattr on the frame instance instead of just taking the values from the dict:

https://github.com/astropy/astropy/blob/main/astropy/io/misc/asdf/tags/coordinates/frames.py#L85

Someone clearly went out of their way to do that, but I don't understand why...